cclib/parser/data.py

```diff
@@ -157,10 +159,15 @@ class ccData:
        "nooccnos":         Attribute(numpy.ndarray,    'TBD',                         'N/A'),
        "nsocoeffs":         Attribute(list,    'TBD',                         'N/A'),
        "nsooccnos":         Attribute(list,    'TBD',                         'N/A'),
+       "oniom_energy":     Attribute(list,             'ONIOM energy',                'optimization:ONIOM'),
+       "oniom_energy_fin": Attribute(float,            'ONIOM energy',                'properties:energy'),
        "optdone":          Attribute(list,             'done',                        'optimization'),
        "optstatus":        Attribute(numpy.ndarray,    'status',                      'optimization'),
        "polarizabilities": Attribute(list,             'polarizabilities',            'N/A'),
        "pressure":         Attribute(float,            'pressure',                    'properties'),
+       "rotemp":           Attribute(numpy.ndarray,    'rotational temperatures',     'properties:rotational'),
+       "spinexpectanni":   Attribute(numpy.ndarray,    'S2 after annihilation',       'optimization'),
+       "spinexpect":       Attribute(numpy.ndarray,    'S2 before annihilation',      'optimization'),

```

Comment:
OK I thought about this and checked some other output files. These are physically the same attribute `spin_squared` or something along those lines. S^2 can be meaningful for different states and different methods so this needs to be multidimensional. Instead of making a list of lists or something lets do a dictionary, `spin_squared['scf'], spin_squared['mp'], spin_squared['cc']`. 

The only thing that is relevant for this PR: Store these in `spin_squared['scf_preannihilation'], spin_squared['scf_postannihilation'], and spin_squared['scf']`. The last one `spin_squared['scf']` should be a duplicate of either `pre` or `post` (IMO `post`) which will ensure that if someone is just grabbing `spin_squared['scf']` from multiple different filetypes then it will just work. 

Does this sound acceptable? 

Other devs please weigh in @oliver-s-lee @ghutchis @langner @berquist @amandadumi

Reply:
This sounds good to me, although we should probably go with `spinsquared` or similar as the attribute name, unless we're moving to underscore convention in future attr names?

Reply:
good point on the naming. `spinsquared` sounds good to me

Reply:
Updated as per our discussion. I've implemented as a list, because presumably we'll get one set of S^2 values per opt cycle.